### DISCUSSION TITLE
"magick.exe: no images found for operation `-trim' at CLI arg 1." The image exists and was passed properly.

### DISCUSSION THREAD

**Discussion by futuremotiondev** [ImageMagick/ImageMagick#6750] [Help]
### ImageMagick version

7.1.1-15 Q16-HDRI x64

### Operating system

Windows

### Operating system, version and so on

Windows 10 Pro x64 Version 22H2 (Build 19045.3448)

### Description

I'm just using a simple `-trim` command:

`magick -trim .\outlook.png outlook-after.png`

But I'm getting the following error:

`magick.exe: no images found for operation `-trim' at CLI arg 1 @ error/operation.c/CLIOption/5470.` 

I'm not understanding why this isn't working?

Any help would be great.

### Steps to Reproduce

1. Place a couple PNGs in a folder.
2. Navigate to that folder in the console (Pwsh / Cmd)
3. Type `magick -trim image-input.png image-output.png` where `image-input.png` is the name of one of the files in the folder.
4. Observe the errror: `magick.exe: no images found for operation `-trim' at CLI arg 1 @ error/operation.c/CLIOption/5470.`

### Images

Screen recording:

![pwsh_JI8cNw6d5M](https://github.com/ImageMagick/ImageMagick/assets/33441569/b2d85b4c-1828-4348-94d8-b9222c9b6c94)

**Comment by snibgo** [comment]
IM executes the operations in the order you give them. `-trim` is an image operation, which can't be done before you have read any images. You should rearrange the command to read the input image, then trim it, then write it.
```
magick .\outlook.png -trim outlook-after.png
```

**Comment by Huy-Ngo** [comment]
Then the instruction in the man page:

```
magick [input‐options] input‐file [output‐options] output‐file
```

is wrong right? input-file should then go before input options

**Comment by snibgo** [comment]
> is wrong right? input-file should then go before input options

No, that isn't wrong. For example:
```
magick -density 300 in.pdf -blur 0,3 out.png
```
Here, `-density 300` is an "input-option". It affects how the file `in.pdf` is rasterized. But `-blur 0x3` is an operation on an image, and it isn't valid to call that operation before any images have been read.

**Reply by Huy-Ngo** [reply]
OK i see
